addons/api/addon/serializers/storage-bucket.js

```diff
@@ -49,17 +49,22 @@ export default class StorageBucketSerializer extends ApplicationSerializer {
         json.attributes[key] = null;
     }
 
+    //json.secrets is only present, if there are any updates to the secret fields
+    if (options.isNestedSecret && json.secrets) {
+      if (!fieldsByCredentialType[credentialType].includes(key)) {
+        delete json.secrets[key];
+      }
```

Comment:
Do we have to worry about having an empty secrets object if we delete all the keys?

Reply:
to be on the safe side - I'm getting rid of the entire secrets now